libs/platform/ebpf_epoch.c

```diff
@@ -10,93 +10,132 @@
 //
 // Each block of code that accesses epoch freed memory wraps access in calls to ebpf_epoch_enter/ebpf_epoch_exit.
 //
-// The per-CPU state is protected by a single per-CPU lock.
+// The per-CPU state is only accessed only by the CPU that owns it and only at IRQL >= DISPATCH_LEVEL.
 //
 // ebpf_epoch_enter:
-// If the thread is preemptible, it blocks waiting for the semaphore (to ensure that the thread entry table has an
-// available entry). If the thread is not preemptible, it is guaranteed that the thread entry table has an available
-// entry. The thread entry table is a fixed size hash table based on the thread ID. The current epoch is read and stored
-// in the thread entry table.
+// First: The IRQL, CPU ID, and current epoch are saved in the ebpf_epoch_state_t.
```

Comment:
Suggest these should be in doxygen @brief for each of the functions mentioned?

Reply:
Done